src/SharedStream.mo

```diff
@@ -7,14 +7,20 @@ import Option "mo:base/Option";
 import Principal "mo:base/Principal";
 import R "mo:base/Result";
 import Time "mo:base/Time";
-import Timer "mo:base/Timer";
+import AssocList "mo:base/AssocList";
+import List "mo:base/List";
 
 import Vec "mo:vector";
 import QueueBuffer "QueueBuffer";
 
 module {
 
-  type StreamError = { #NotRegistered; #StreamClosed };
+  public type ChunkError = {
+    #BrokenPipe : (expectedIndex : Nat, receivedIndex : Nat);
```

Comment:
We can remove the `receivedIndex` component here. The sender already knows that index from his own call context when processing the response.

Reply:
removed